### PR TITLE
feat: 1674 Auto-hide sidebar when re-sizing the window

### PR DESCRIPTION
**Pull request by Shreesh09**
[FR] #1674 

https://github.com/AppFlowy-IO/AppFlowy/assets/114166000/88dab107-c489-4786-bdff-7e93f34eb81d

### Feature Preview
Sidebar will automatically collapse if the window size is small and will reappear when the window is large enough

---

#### PR Checklist

- [x] My code adheres to [AppFlowy's Conventions](https://docs.appflowy.io/docs/documentation/software-contributions/conventions)
- [x] I've listed at least one issue that this PR fixes in the description above.
- [ ] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [x] All existing tests are passing.

### CODE REVIEW

File: frontend/appflowy_flutter/lib/workspace/presentation/home/home_layout.dart [hunk 1]
```diff
@@ -29,6 +29,11 @@ class HomeLayout {
 
     menuWidth += homeSetting.resizeOffset;
 
+    final screenWidthPx = context.widthPx;
+    context
+        .read<HomeSettingBloc>()
+        .add(HomeSettingEvent.screenSmall(screenWidthPx));
```
[the remark this answers is not in the document]
Comment (Shreesh09) [reply frontend/appflowy_flutter/lib/workspace/presentation/home/home_layout.dart:35]: I'm sorry for any confusion. I'll rename it to something more fitting.

The purpose of that event is to monitor changes in screen size (e.g., when the screen transitions from large to small or vice versa) and determine whether to collapse the menu or not.

**Review by richardshiue — APPROVED** [review]
LGTM

**Comment by LucasXu0** [comment]
@richardshiue @Xazin  Almost all the integration tests failed. We should not merge this PR.

**Comment by richardshiue** [comment]
Sorry, I messed up. I've submitted a fix here: https://github.com/AppFlowy-IO/AppFlowy/pull/3673/files

**Pull request merged**